### DISCUSSION TITLE
AWS Machine Image Custom Versioning

### DISCUSSION THREAD

**Discussion by knechtionscoding** [renovatebot/renovate#23421] [Request Help]
### How are you running Renovate?

Placeholder value, please select the correct response from the dropdown

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.122.0

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I'd like to use git tags to do lookups for AMIs rather than AWS so that I can do things like provide compatibility and not need to grant access to AWS to renovate when we publish and build the AMIs from git anyway.

### What you tried so far.

So far I've tried:

```json
            {
              fileMatch: ['.*\\.tf$'],
              "depNameTemplate": "<repo>",
              [
              ".*image_tag = (?<compatability>(?:(?!ami).)*)(?<currentValue>ami-[a-z0-9]{17}).*",
              ],
              "datasourceTemplate": "git-tags",
              "versioningTemplate": "aws-machine-image",
              "registryUrls": ["<git-url>"]
            }
```

However this doesn't work because the versioning isn't just raw `ami-1111111111111` which means I can't use the versioningTemplate `aws-machine-image`. However I also can't use the regex versioning because it requires the major/minor/patch configuration of semver and ami's don't follow that.

Thoughts on the best way to move forward?

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by knechtionscoding** [comment]
@rarkins curious if you have any ideas here on the best way to solve this?

**Reply by rarkins** [reply]
I don't understand this requirement sorry. Maybe you can create a reproduction repo to demonstrate. Regex cersionig does not require minor and patch, alternatively "loose" versioning works on any numbers

**Reply by knechtionscoding** [reply]
Apologies I've been traveling and out of office, let me rephrase.

We'd like to get AMI versions from git-tags rather than AWS. This means the versioning has to support the format: ami-XXXXXXXXX.

The regex manager requires a major version, but there's no guarantee that the AMI ID is sequential, and loose might work, if we key off of the date the image is published?

Fundamentally, we want to write a parser/manager that supports the following version: `<machine-type>-<operating-system>-<date-time>-<ami-id>`. The AMI-ID is what should be put into the terraform code, The first two are compatibility parts, and the third obviously, date-time of publish (which could be the "major version"?)
